### ISSUE TITLE
Force push to master

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#2507]
Comparing the [closed pull requests](https://github.com/eslint/eslint/pulls?q=is%3Apr+is%3Aclosed+sort%3Aupdated-desc) with the commit history on master, it seems like there is d90c281 and eb5860c missing.

@gyandeeps According to your [public activity](https://github.com/gyandeeps?tab=activity) you pushed to master.

**Comment by gyandeeps** [comment]
Dang it. I am so sorry for this.
Git on my machine is being acting weird for past 1 month. It has started doing this detach on the rebase. I read that it will do that but you have to do a force push to push your changes.  
I did a force push on my branch but I am not sure why it pushed it to master. And I have does this in past.

**Comment by nzakas** [comment]
Can one of you cherry pick those commits back to master?

@gyandeeps you might want to do this:

    git config --global push.default current

That makes sure that `git push` always good to a remote branch of the same name.

**Comment by gyandeeps** [comment]
@nzakas Thanks for this update. I will make that change and also now on I will work on my own fork of eslint till the time I become comfortable with `git`. Sorry about that.

@lo1tuma Can you please perform the cherry pick as I am not the best person to do that. Thanks aand sorry about the trouble.

**Comment by lo1tuma** [comment]
Done, the commits are on master again (f9d96a54b3ae2dd55d324446ec7aafddd3a2d02e and f20fd419505f989a2256f88e0587ed89229aa7b3).

Anything we can do to prevent this in the future? AFAIK you can forbid force pushes to the default branch on GitHub Enterprise and for github.com you can ask the support to make this setting for a specific repository.

**Comment by gyandeeps** [comment]
Thanks a lot @lo1tuma .

**Comment by hzoo** [comment]
Maybe a `pre-push` hook that prevents force pushes? I guess that would be a local thing though.

**Comment by gyandeeps** [comment]
Just a question, when you guys do a `rebase` to squash your commit into 1 on your branch. Do you use `force push` always?
Is there a different way for pushing it in as when I rebase it always leads to detached HEAD and that time you can't do a normal push?

**Comment by hzoo** [comment]
Yeah since the history is different now (no way to fast-forward it)- you are basically overwriting the multiple commits on the remote branch with the single squashed commit - I do `git rebase -i`, pick commits to squash, check the log, then `git push -f`.

**Comment by gcochard** [comment]
If I don't have a PR open already, I always push to a new branch, so I rebase then `git push origin <oldbranch>:<newbranch>`. If I am updating an existing PR, I will force push. I have `push.default` set to `simple` as well.

**Comment by ilyavolodin** [comment]
@gyandeeps if you already have PR then you always have to do force push for any alteration of the commit (git rebase -i, git amend, etc.) I ran into the same exact thing very early on, when I just started using git, and from then on, I adopted a rule to always specify full information for any commit I do. So instead of doing `git push` or `git pull`, I always do `git push origin name_of_the_branch` and `git pull origin master`. Just to make sure that at any given time, I know exactly what my command is going to do.
